posts/inside-rust/2024-01-23-crates-io-typosquatting-retrospective.md

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: "A crates.io typosquatting retrospective"
+author: Adam Harvey
+team: the crates.io team <https://www.rust-lang.org/governance/teams/crates-io>
+---
+
+About two months ago, the crates.io team [agreed to run an experiment][typomania-pr] for about two months where all new crates would be checked for potential typosquatting as they were published, after which a decision would be made on whether these checks had sufficient value to retain them.
+
+When potential typosquatting was found during this experiment, two members of Rust Foundation staff were notified: myself, and Walter Pearce[^security-initiative]. When notified, one or both of us checked the crate, and then malicious crates had appropriate action[^action] taken in collaboration with the [crates.io team][crates-io-team] and the [security response working group][security-response-wg].
+
+## Results
+
+Between the deployment of the typosquatting checks on November 14, 2023, and the writing of this blog on January 22, 2024, 6,799 new crates were published. 106 (1.6%) triggered one or more typosquatting checks, of which 2 (1.9% of those crates) turned out to be malicious. We also found an additional 2 malicious crates as a result of these checks that did not trigger typosquatting checks[^also].
```

Comment:
> We also found an additional 2 malicious crates as a result of these checks that did not trigger typosquatting checks

if you've found them "as a result of these checks" then how could they "not trigger typosquatting checks"? 😅

Reply:
I guess the answer is in the footnotes, but maybe it makes more sense to explain it in the text itself :D